src/python/pants/backend/python/goals/run_helper.py

```diff
@@ -83,7 +83,12 @@ async def _create_python_source_run_request(
 
     pex_request = dataclasses.replace(pex_request, pex_path=(*pex_request.pex_path, *pex_path))
 
-    complete_pex_environment = pex_env.in_workspace()
+    if run_in_sandbox:
+        # Note that a RunRequest always expects to run directly in the sandbox/workspace
+        # root, hence working_directory=None.
+        complete_pex_environment = pex_env.in_sandbox(working_directory=None)
+    else:
+        complete_pex_environment = pex_env.in_workspace()
```

Comment:
I'm rereading this and I still only get like 80% of what's going on here. Don't burn CI, but if you can think of a different/better way of wording...

Actually if anything the comment might just be the confusing bit. The code is clear without it (to me at least) "if we're running in the sandbox, use a sandbox env, otherwise use a workspace env)

Reply:
Yeah, the comment is to explain why we set `working_directory=None`. I will reword but skip CI.

Reply:
Although, I can't really think of a rewording. But yeah, hopefully the larger question is clear from the code, as you state.